ground/src/main/java/com/google/android/ground/ui/map/gms/GoogleMapsFragment.kt

```diff
@@ -0,0 +1,454 @@
+    val marker =
+      getMap()
+        .addMarker(
+          MarkerOptions()
+            .position(position)
+            .icon(getMarkerIcon(mapPin))
+            .anchor(0.5f, 0.85f)
+            .alpha(1.0f)
+        )
+    markers.add(marker)
+    marker.tag = mapPin
+  }
+
+  private fun getMarkerIcon(mapPin: MapPin): BitmapDescriptor =
+    markerIconFactory.getMarkerIcon(parseColor(mapPin.style.color), currentZoomLevel)
+
+  private fun addMultiPolygon(locationOfInterest: MapLocationOfInterest, multiPolygon: MultiPolygon) =
+    multiPolygon.polygons.forEach { addPolygon(locationOfInterest, it) }
+
+  private fun addPolygon(locationOfInterest: MapLocationOfInterest, polygon: Polygon) {
+    val options = PolygonOptions()
+    options.clickable(false)
+    val shellVertices = polygon.shell.vertices.map { toLatLng(it) }
+    options.addAll(shellVertices)
+    val holes =
+      polygon.holes.map { hole -> hole.vertices.map { point -> toLatLng(point) } }
+    holes.forEach { options.addHole(it) }
+
+    val mapsPolygon = getMap().addPolygon(options)
+    mapsPolygon.tag = Pair(locationOfInterest.locationOfInterest!!.id, LocationOfInterest::javaClass)
+    mapsPolygon.strokeWidth = polylineStrokeWidth.toFloat()
+    // TODO(jsunde): Figure out where we want to get the style from
+    //  parseColor(Style().color)
+    mapsPolygon.fillColor = parseColor("#55ffffff")
+    mapsPolygon.strokeColor = parseColor(Style().color)
+    mapsPolygon.strokeJointType = JointType.ROUND
+
+    polygons.getOrPut(locationOfInterest) { mutableListOf()} .add(mapsPolygon)
+  }
+
+  private val polylineStrokeWidth: Int
+    get() = resources.getDimension(R.dimen.polyline_stroke_width).toInt()
+
+  private fun onMapClick(latLng: LatLng) = handleAmbiguity(latLng)
+
+  override fun getCurrentZoomLevel(): Float = getMap().cameraPosition.zoom
+
+  @SuppressLint("MissingPermission")
+  override fun enableCurrentLocationIndicator() {
+    if (!getMap().isMyLocationEnabled) {
+      getMap().isMyLocationEnabled = true
+    }
+  }
+
+  override fun setMapLocationsOfInterest(features: ImmutableSet<MapLocationOfInterest>) {
+    Timber.v("setMapLocationsOfInterest() called with ${features.size} locations of interest")
+    val features: MutableSet<MapLocationOfInterest> = HashSet(features)
+
+    //    val deletedMarkers: MutableList<Marker> = ArrayList()
+    //    for (marker in markers) {
+    //      val pin = marker.tag as MapPin
+    //      if (features.contains(pin)) {
+    //        // If existing pin is present and up-to-date, don't update it.
+    //        geometriesToUpdate.remove(pin)
+    //      } else {
+    //        // If pin isn't present or up-to-date, remove it so it can be added back later.
+    //        removeMarker(marker)
+    //        deletedMarkers.add(marker)
+    //      }
+    //    }
+
+    //    // Update markers list.
+    //    deletedMarkers.forEach { o: Marker -> markers.remove(o) }
```

Comment:
Is this ok to be cleaned?

Reply:
I just looked at the original class, and looks like this is not implemented. Please confirm